src/block-components/helpers/size/style.js

```diff
@@ -123,6 +124,10 @@ const getStyleParams = ( options = {} ) => {
 			attrNameTemplate,
 			responsive: 'all',
 			hasUnits: 'px',
+			enabledCallback: getAttribute => {
+				const deviceType = getDeviceType()
```

Comment:
This will not work as intended because if you try and get the `deviceType` here, you will just end up getting the _current_ device type/preview of the editor (not the device type of the style being currently generated). Unfortunately the `device` isn't provided in the `enabledCallback` arguments.

I think we should try an alternative approach.. instead of `enabledCallback`, try using the `valuePreCallback`.. that callback is used to override the value of the attribute before it gets processed for style generation. If the function `valuePreCallback` returns an empty string `''`, the style isn't generated.